packages/flutter/lib/src/material/input_decorator.dart

```diff
@@ -40,6 +40,7 @@ class _InputBorderGap extends ChangeNotifier {
   }
 
   @override
+  // ignore: avoid_equals_and_hash_code_on_mutable_classes
```

Comment:
This one seems odd. Why is this one allowed to be mutable, but still have an overridden operator==? @HansMuller 

We should add a comment why this ignore is legit (see style guide).

Reply:
_InputBorderPainter.shouldRepaint() compares _InputBorderGap values with == but _InputBorderGap is not used in a collection

Reply:
That's not really the same thing. On the other hand this is a small internal class and the hashCode override my serve as a warning to someone in the future, so OK.